src/redux/sagas/expenditures/createStreamingPayment.ts

```diff
@@ -103,6 +104,43 @@ function* createStreamingPaymentAction({
       BigNumber.from(10).pow(getTokenDecimalsWithFallback(tokenDecimals)),
     );
 
+    let realEndTimestamp: BigNumber;
+
+    switch (endCondition) {
+      case StreamingPaymentEndCondition.FixedTime:
+        if (endTimestamp === undefined) {
+          throw new Error(
+            'endTimestamp is required for FixedTime endCondition',
+          );
+        }
+
+        realEndTimestamp = BigNumber.from(endTimestamp);
+        break;
+
+      case StreamingPaymentEndCondition.LimitReached:
+        if (limitAmount === undefined) {
+          throw new Error(
+            'limitAmount is required for LimitReached endCondition',
+          );
+        }
+
+        realEndTimestamp = BigNumber.from(limitAmount ?? 0).eq(0)
+          ? BigNumber.from(startTimestamp)
+          : BigNumber.from(limitAmount ?? 0)
+              .mul(interval)
+              .div(amount)
+              .add(startTimestamp);
+        break;
+
+      case StreamingPaymentEndCondition.WhenCancelled:
+        realEndTimestamp = TIMESTAMP_IN_FUTURE;
+        break;
+
+      default:
+        realEndTimestamp = TIMESTAMP_IN_FUTURE;
+        break;
+    }
+
```

Comment:
Perhaps this could live in a helper so it can also be reused by the motion version of the saga?

Reply:
Actually, will be easier if I move it in #2212

Reply:
Agree, this should be extracted